---
identifier: "CVE-2021-41212"
identifiers:
- "CVE-2021-41212"
- "GHSA-fr77-rrx3-cp7g"
package_slug: "pypi/tensorflow"
title: "Out-of-bounds Read"
description: "In affected versions the shape inference code for `tf.ragged.cross` can 
  trigger a read outside of bounds of heap allocated array."
date: "2021-11-09"
pubdate: "2021-11-05"
affected_range: "<2.4.4||>=2.5.0,<2.5.2||==2.6.0"
fixed_versions:
- "2.4.4"
- "2.5.2"
- "2.6.1"
affected_versions: "All versions before 2.4.4, all versions starting from 2.5.0 before
  2.5.2, version 2.6.0"
not_impacted: "All versions starting from 2.4.4 before 2.5.0, all versions starting
  from 2.5.2 before 2.6.0, all versions after 2.6.0"
solution: "Upgrade to versions 2.4.4, 2.5.2, 2.6.1 or above."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2021-41212"
- "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-fr77-rrx3-cp7g"
- "https://github.com/tensorflow/tensorflow/commit/fa6b7782fbb14aa08d767bc799c531f5e1fb3bb8"
cvss_v2: "AV:L/AC:L/Au:N/C:P/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:H"
uuid: "77bb6d61-ab46-4bbe-ab13-9001796b7b08"
cwe_ids:
- "CWE-1035"
- "CWE-125"
- "CWE-937"
